core/src/test/java/org/bitcoinj/core/PeerTest.java

```diff
@@ -94,7 +94,7 @@ public void tearDown() throws Exception {
     }
 
     private void connect() throws Exception {
-        connectWithVersion(70001, VersionMessage.NODE_NETWORK);
+        connectWithVersion(70012, VersionMessage.NODE_NETWORK | VersionMessage.NODE_WITNESS);
```

Comment:
This change changes all the tests. I wonder if we should configure this class such that tests are run twice, once with witness/WitnessBlock and once without?

Reply:
Is the node itself ever expected to have the `NODE_WITNESS` flag disabled after the fork?

Reply:
I can't answer this question for sure, but I assume yes. Bitcoinj is used in networks that don't have segwit (yet). Also even in the Bitcoin P2P network peer versions <70012 will exist for quite some years.

One also could argue that simply because we have the code/ability to do so we should continue to have tests for it. It seems to me that we now have quite some `if (segwit) { … } else { … }` in code, and it feels wrong to me to drop tests we already have for the else branches.